### PR TITLE
MicroProfile Language Support for Open Liberty blog post

### PR DESCRIPTION
**Pull request by kathrynkodama**
For issue #774

### CODE REVIEW

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+author_picture: https://avatars3.githubusercontent.com/kathrynkodama
+author_github: https://github.com/kathrynkodama
+seo-title: MicroProfile Language Support for Open Liberty Applications - OpenLiberty.io
+seo-description: Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty.
+blog_description: "Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty."
+---
+= MicroProfile Language Support for Open Liberty Applications
+Kathryn Kodama https://github.com/kathrynkodama
+
+The new MicroProfile Language Server provides added support for developing your Eclipse MicroProfile applications with Open Liberty. Earlier this month, the first release of the Eclipse MicroProfile Language Server was published as a joint effort between IBM and Red Hat. The Tools for MicroProfile VS Code extension consumes this language server, providing feature-rich support for MicroProfile APIs and properties in VS Code.
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]:
```suggestion
The new MicroProfile Language Server provides added support for developing your Eclipse MicroProfile applications with Open Liberty. Earlier this month, the first release of the Language Server for Eclipse MicroProfile (Eclipse LSP4MP), which is an open incubator project under the Eclipse Foundation, was published. The Tools for MicroProfile VS Code extension consumes this language server, providing feature-rich support for MicroProfile APIs and properties in VS Code.
```

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+- https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Open Liberty Tools]: Support for development with Open Liberty.
+
+- https://marketplace.visualstudio.com/items?itemName=MicroProfile-Community.mp-starter-vscode-ext[MicroProfile Starter]: Quickly generate a MicroProfile Maven project.
+
+== Setting up the project
+
+=== Create a new MicroProfile project
+
+Generate a new Open Liberty Maven application with the MicroProfile Starter.
```
Comment (rzgry) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]: Maybe we want to include some quick information on how to use the MP Starter extension as people may not know how.

Ie. Open Command Pallet (Cmd + Shift + P) -> `MicroProfile: Generate a new MicroProfile Starter Project` -> Enter package name + artifact id -> Select MP 3.3 -> Select Open Liberty -> Select Java Version -> Select the specs you want to use.

Or maybe we could include a gif of project generation?

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+seo-title: MicroProfile Language Support for Open Liberty Applications - OpenLiberty.io
+seo-description: Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty.
+blog_description: "Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty."
+---
+= MicroProfile Language Support for Open Liberty Applications
+Kathryn Kodama https://github.com/kathrynkodama
+
+The new MicroProfile Language Server provides added support for developing your Eclipse MicroProfile applications with Open Liberty. Earlier this month, the first release of the Eclipse MicroProfile Language Server was published as a joint effort between IBM and Red Hat. The Tools for MicroProfile VS Code extension consumes this language server, providing feature-rich support for MicroProfile APIs and properties in VS Code.
+
+In this blog, we will showcase how the new MicroProfile Language Server, along with existing tooling for VS code, makes it even easier to develop your Open Liberty applications.
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]:
```suggestion
In this blog, we will showcase how the new MicroProfile Language Server, along with existing tooling for VS Code, makes it even easier to develop your Open Liberty applications.
```

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 1 above)
Comment (yeekangc) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: If we want to note the participants of the Eclipse project (e.g. IBM, RH, the MP community), we can still.  Then, we will note them under the context of the Eclipse project.

@kathrynkodama @lauracowen ^^^

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+[source, xml]
+----
+<featureManager>
+    <feature>jaxrs-2.1</feature>
+    <feature>jsonp-1.1</feature>
+    <feature>cdi-2.0</feature>
+    <feature>mpMetrics-2.0</feature>
+    <feature>mpConfig-1.3</feature>
+    <feature>mpHealth-2.2</feature>
+</featureManager>
```
Comment (rzgry) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]: Do we recommend people enable features individually as needed, or do we recommend users just use the microprofile convenience features. @yeekangc any thoughts on this? 
```
<featureManager>
  <feature>microProfile-3.3</feature>
<featureManager>
```

**Review by yeekangc — CHANGES_REQUESTED** [review]

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+----
+
+== Starting dev mode
+
+Start your Open Liberty application in dev mode using the Liberty Dev Dashboard, or via command line using `mvn liberty:dev`/`gradle libertyDev`. Dev mode provides hot reload and deployment, so your code changes will automatically be reflected in your running application.
```
Comment (rzgry) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]: Should we mention liberty dev dashboard is part of vscode? maybe something like:  

Start your Open Liberty application in dev mode using vscode with the Liberty Dev Dashboard, or via command line using?

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+[.img_border_light]
+image::/img/blog/liberty-dev-dashboard-vscode.png[Liberty Dev Dashboard in VS Code]
+
+== Developing with MicroProfile language support
+
+The MicroProfile Language Server provides useful features such as snippets, completion, hover, and diagnostics for both Java and properties files.
+
+=== Coding assistance with Java source files
+
+Use a snippet to quickly insert popular code blocks. For example,  the  `mpreadiness` snippet inserts a fully functional MicroProfile Health Check Readiness class. Open Liberty dev mode will automatically pick up any changes and recompile your source code so that you can immediately access your new health class at the `/health` endpoint.
```
Comment (rzgry) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]: I think it is just "MicroProfile Health" so maybe change to "inserts a fully functional MicroProfile Health Readiness check class"

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+[.img_border_light]
+image::/img/blog/mp-readiness-snippet.gif[MicroProfile readiness snippet]
+
+The MicroProfile Language Server has knowledge of the MicroProfile APIs, meaning your source code will validate and correct improper use of MicroProfile features. For example, as per the MicroProfile Health specification, all MicroProfile Health classes using the `@Readiness` or `@Liveness` annotations must implement the `HealthCheck` class. The MicroProfile Language Server will correct you where needed with a helpful diagnostic.
```
Comment (rzgry) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]: Is this better? 

The MicroProfile Language Server has knowledge of the MicroProfile APIs, meaning the language server will validate and correct improper use of MicroProfile APIs in your source code.

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 1 above)
Comment (kathrynkodama) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: Updated with the wording you suggested.  I don't feel we need to note the participants, just that it is an open incubator project under the Eclipse Foundation is sufficient.

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 4 above)
Comment (yeekangc) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: It depends on what the users want to accomplish and what we want to accomplish here.  Individual features will give you optimal footprint.

For our purpose here, it's probably alright to use the convenience feature for brevity and simplicity (with a note about that).

Comment (kathrynkodama) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: Updated to:
```
Add the MicroProfile features you would like to use in your Open Liberty `server.xml` file. 
For convenience, we have specified `microprofile-3.3` to include all of the MicroProfile features.

[source, xml]
----
<featureManager>
    <feature>microProfile-3.3</feature>
</featureManager>
----
```

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 8]
```diff
@@ -0,0 +1,158 @@
+---
+layout: post
+title: "MicroProfile Language Support for Open Liberty Applications"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kathrynkodama
+author_github: https://github.com/kathrynkodama
+seo-title: MicroProfile Language Support for Open Liberty Applications - OpenLiberty.io
+seo-description: Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty.
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]:
```suggestion
seo-description: Leverage the new MicroProfile Language Server along with Visual Studio Code tools to develop your cloud-native Java applications with Eclipse MicroProfile API and Open Liberty.
```

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+blog_description: "Leverage the new MicroProfile Language Server along with existing VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty."
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]:
```suggestion
blog_description: "Leverage the new MicroProfile Language Server along with VS Code tools to develop your Eclipse MicroProfile applications with Open Liberty."
```

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+== Installing the extensions
+
+The following extensions are available on the VS Code Marketplace to support developing your Open Liberty applications with MicroProfile.
+
+- https://marketplace.visualstudio.com/items?itemName=redhat.vscode-microprofile[Tools for MicroProfile]: Language support for MicroProfile APIs and properties via the https://github.com/eclipse/lsp4mp[Language Server for Eclipse].
+
+- https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Open Liberty Tools]: Support for development with Open Liberty.
+
+- https://marketplace.visualstudio.com/items?itemName=MicroProfile-Community.mp-starter-vscode-ext[MicroProfile Starter]: Quickly generate a MicroProfile Maven project.
+
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc:27]: Offer an option for users to install all in one go with MicroProfile Extension Pack?

**Review by yeekangc — DISMISSED** [review]

**Comment by yeekangc** [comment]
@kathrynkodama, please address the last few comments.  I think we are good to go after.  Thanks.

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 8 above)
Comment (kathrynkodama) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: Updated seo-description

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 9 above)
Comment (kathrynkodama) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc]: Updated blog description

File: posts/2020-09-30-microprofile-language-support-liberty.adoc (hunk 10 above)
Comment (kathrynkodama) [reply posts/2020-09-30-microprofile-language-support-liberty.adoc:27]: Updated to: 

Install the https://marketplace.visualstudio.com/items?itemName=MicroProfile-Community.vscode-microprofile-pack[MicroProfile extension pack] from the VS Code Marketplace to support developing your Open Liberty applications with MicroProfile. We will use the following extensions from the MicroProfile extension pack:

File: posts/2020-09-30-microprofile-language-support-liberty.adoc [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+    }
+}
+----
+
+Add the MicroProfile dependency to your `build.gradle` file.
+
+[source,groovy]
+----
+dependencies {
+    providedCompile 'org.eclipse.microprofile:microprofile:3.2'
```
Comment (yeekangc) [inline posts/2020-09-30-microprofile-language-support-liberty.adoc]:
```suggestion
    providedCompile 'org.eclipse.microprofile:microprofile:3.3'
```

**Review by yeekangc — DISMISSED** [review]

**Review by yeekangc — APPROVED** [review]

**Pull request merged**